lldb/docs/resources/debugging-lldb.rst

```diff
@@ -0,0 +1,254 @@
+Debugging LLDB
+==============
+
+This page details various ways to debug LLDB itself and other LLDB tools. If
+you want to know how to use LLDB in general, please refer to
+:doc:`/use/tutorial`.
+
+As LLDB is generally split into 2 tools, ``lldb`` and ``lldb-server``
+(``debugserver`` on Mac OS), the techniques shown here will not always apply to
+both. With some knowledge of them all, you can mix and match as needed.
+
+In this document we refer to the initial ``lldb`` as the debugger and the
+program being debugged as the debugee.
+
+Building For Debugging
+----------------------
+
+To build LLDB with debugging information add the following to your CMake
+configuration:
+
+::
+
+  -DCMAKE_BUILD_TYPE=Debug \
+  -DLLDB_EXPORT_ALL_SYMBOLS=1
```

Comment:
I would also recommend explicitly turning on `LLVM_ENABLE_ASSERTIONS=On` as well -- Some bugs may be easier to diagnose if LLDB asserts instead of returning an error somewhere.

Reply:
It's on by default for Debug but there are ways to debug using release builds, so I've noted it in that context.